File: packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/Role.kt
/**
 * Copyright 2020, Yahoo Holdings Inc.
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */
package com.yahoo.navi.ws.models.beans

import com.yahoo.elide.annotation.Include
import com.yahoo.elide.annotation.SharePermission
import com.yahoo.elide.annotation.DeletePermission
import com.yahoo.elide.annotation.CreatePermission
import com.yahoo.elide.annotation.UpdatePermission
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp

import java.util.Date

import javax.persistence.Entity
import javax.persistence.Table
import javax.persistence.Id
import javax.persistence.Column
import javax.persistence.Temporal
import javax.persistence.TemporalType
import javax.validation.constraints.NotBlank

@Entity
@Table(name = "navi_roles")
@Include(rootLevel = true, type = "roles")
@SharePermission
@DeletePermission(expression = "everybody")
@CreatePermission(expression = "everybody")
@UpdatePermission(expression = "nobody")
class Role {
    @get:Id
    @get:NotBlank
    var id: String? = null

    @get:CreationTimestamp
    @get:Column(columnDefinition = "timestamp default current_timestamp")
    @get:Temporal(TemporalType.TIMESTAMP)
    @get:UpdatePermission(expression = "nobody")
    var createdOn: Date? = null

    @get:UpdateTimestamp
    @get:Column(columnDefinition = "timestamp default current_timestamp")
    @get:Temporal(TemporalType.TIMESTAMP)
    @get:UpdatePermission(expression = "nobody")
    var updatedOn: Date? = null
}

File: packages/core/addon/models/role.ts
/**
 * Copyright 2021, Yahoo Holdings Inc.
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */
import attr from 'ember-data/attr';
import Model from '@ember-data/model';
import { Moment } from 'moment';

export default class Role extends Model {
  @attr('moment') createdOn!: Moment;
  @attr('moment') updatedOn!: Moment;
}

// DO NOT DELETE: this is how TypeScript knows how to look up your models.
declare module 'ember-data/types/registries/model' {
  export default interface ModelRegistry {
    role: Role;
  }
}
